specification/cdn/resource-manager/Microsoft.Cdn/preview/2025-07-01-preview/cdnwebapplicationfirewall.json

```diff
@@ -0,0 +1,952 @@
+        "policySettings": {
+          "description": "Describes  policySettings for policy",
+          "$ref": "#/definitions/policySettings"
+        },
+        "rateLimitRules": {
+          "description": "Describes rate limit rules inside the policy.",
+          "$ref": "#/definitions/RateLimitRuleList"
+        },
+        "customRules": {
+          "description": "Describes custom rules inside the policy.",
+          "$ref": "#/definitions/CustomRuleList"
+        },
+        "managedRules": {
+          "description": "Describes managed rules inside the policy.",
+          "$ref": "#/definitions/ManagedRuleSetList"
+        },
+        "endpointLinks": {
+          "description": "Describes Azure CDN endpoints associated with this Web Application Firewall policy.",
+          "type": "array",
+          "readOnly": true,
+          "items": {
+            "$ref": "#/definitions/CdnEndpoint"
+          }
+        },
+        "extendedProperties": {
+          "description": "Key-Value pair representing additional properties for Web Application Firewall policy.",
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          }
+        },
+        "provisioningState": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Provisioning state of the WebApplicationFirewallPolicy.",
+          "enum": [
+            "Creating",
+            "Succeeded",
+            "Failed"
+          ],
+          "x-ms-enum": {
+            "name": "ProvisioningState",
+            "modelAsString": true
+          }
+        },
+        "resourceState": {
+          "title": "Resource status of the policy.",
+          "readOnly": true,
+          "enum": [
+            "Creating",
+            "Enabling",
+            "Enabled",
+            "Disabling",
+            "Disabled",
+            "Deleting"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "PolicyResourceState",
+            "modelAsString": true
+          }
+        }
+      }
+    },
+    "policySettings": {
+      "description": "Defines contents of a web application firewall global configuration",
+      "properties": {
+        "enabledState": {
+          "description": "describes if the policy is in enabled state or disabled state",
+          "type": "string",
+          "enum": [
+            "Disabled",
+            "Enabled"
+          ],
+          "x-ms-enum": {
+            "name": "PolicyEnabledState",
+            "modelAsString": true
+          }
+        },
+        "mode": {
+          "description": "Describes if it is in detection mode or prevention mode at policy level.",
+          "type": "string",
+          "enum": [
+            "Prevention",
+            "Detection"
+          ],
+          "x-ms-enum": {
+            "name": "PolicyMode",
+            "modelAsString": true
+          }
+        },
+        "defaultRedirectUrl": {
```

Comment:
>defaultRedirectUrl [](http://example.com/codeflow?start=9&length=18)

(nit, want to make it format uri?)

Reply:
This file is preexisting and unrelated to the API changes currently under review. I will pass feedback along to the feature owners.